tsdb/cmd/tsdb/main.go

```diff
@@ -653,3 +702,107 @@ func dumpSamples(db *tsdb.DBReadOnly, mint, maxt int64) (err error) {
 	}
 	return nil
 }
+
+// go run cmd/tsdb/main.go dump ../../snapshots/snapshots/20191017T205052Z-32c83bebdee150eb --label-name=__name__ --label-value=etcd_disk_wal_fsync_duration_seconds_sum --format=sqlite3
+func dumpSamplesSqlite3(db *tsdb.DBReadOnly, cfg *dumpConfiguration) (err error) {
+	dbSqlite3, err := sql.Open("sqlite3", cfg.sourceName)
+	if err != nil {
+		return err
+	}
+	defer dbSqlite3.Close()
+
+	sqlStmt := `
+	CREATE TABLE IF NOT EXISTS labels (
+	  id INTEGER PRIMARY KEY,
+	  labels TEXT NOT NULL,
+	  UNIQUE(labels) ON CONFLICT IGNORE
+	);
+	`
+	_, err = dbSqlite3.Exec(sqlStmt)
+	if err != nil {
+		return fmt.Errorf("error creating labels table: %v", err)
+	}
+
+	sqlStmt = `
+	CREATE TABLE IF NOT EXISTS data (
+	  label_id INTEGER NOT NULL,
+	  timestamp INTEGER NOT NULL,
+	  value REAL NOT NULL,
+	  FOREIGN KEY(label_id) REFERENCES labels(id)
+	);
+	`
+	_, err = dbSqlite3.Exec(sqlStmt)
+	if err != nil {
+		return fmt.Errorf("error creating data table: %v", err)
+	}
+
+	tx, err := dbSqlite3.Begin()
+	if err != nil {
+		return fmt.Errorf("error beginning transaction: %v", err)
+	}
+	stmt, err := tx.Prepare("INSERT INTO labels(labels) values(?)")
+	if err != nil {
+		return fmt.Errorf("error preparing transaction: %v", err)
+	}
+	defer stmt.Close()
+
+	stmtData, err := tx.Prepare("INSERT INTO data(label_id, timestamp, value) values((SELECT id FROM labels WHERE labels = ?), ?, ?);")
```

Comment:
this subquery is probably going to be slow. But you can merge this, we can optimize later if it's too slow.